src/overmap_ui.cpp

```diff
@@ -1650,14 +1650,25 @@ static tripoint_abs_omt display( const tripoint_abs_omt &orig,
             ptype.only_known_by_player = true;
             ptype.avoid_danger = true;
             avatar &player_character = get_avatar();
-            bool in_vehicle = player_character.in_vehicle && player_character.controlling_vehicle;
             map &here = get_map();
-            const optional_vpart_position vp = here.veh_at( player_character.pos() );
-            if( vp && in_vehicle ) {
-                vehicle &veh = vp->vehicle();
-                if( veh.can_float() && veh.is_watercraft() && veh.is_in_water() ) {
-                    ptype.only_water = true;
-                } else if( veh.is_rotorcraft() && veh.is_flying_in_air() ) {
+            bool driving = player_character.in_vehicle && player_character.controlling_vehicle;
+            vehicle *player_veh = nullptr;
+            if( driving ) {
+                const optional_vpart_position vp = here.veh_at( player_character.pos() );
+                if( !vp.has_value() ) {
+                    debugmsg( "Failed to find driven vehicle" );
+                    continue;
+                }
+                player_veh = &vp->vehicle();
+            }
+            if( driving ) {
```

Comment:
```suggestion
```
Unless driving is (potentially) getting changed in some way I'm not seeing?

Reply:
Ah, yes. I don't know how that happened.